packages/platform-ios/src/config/index.ts

```diff
@@ -5,86 +5,67 @@
  * LICENSE file in the root directory of this source tree.
  *
  */
-
 import path from 'path';
-import {memoize} from 'lodash';
-import findProject from './findProject';
+import fs from 'fs';
 import findPodfilePath from './findPodfilePath';
+import findXcodeProject from './findXcodeProject';
 import findPodspec from './findPodspec';
 import {
   IOSProjectParams,
   IOSDependencyParams,
+  IOSProjectConfig,
 } from '@react-native-community/cli-types';
 
-const memoizedFindProject = memoize(findProject);
-
-/**
- * For libraries specified without an extension, add '.tbd' for those that
- * start with 'lib' and '.framework' to the rest.
- */
-const mapSharedLibaries = (libraries: Array<string>) =>
-  libraries.map((name) => {
-    if (path.extname(name)) {
-      return name;
-    }
-    return name + (name.indexOf('lib') === 0 ? '.tbd' : '.framework');
-  });
-
 /**
  * Returns project config by analyzing given folder and applying some user defaults
  * when constructing final object
  */
-export function projectConfig(folder: string, userConfig: IOSProjectParams) {
+export function projectConfig(
+  folder: string,
+  userConfig: IOSProjectParams,
+): IOSProjectConfig | null {
   if (!userConfig) {
-    return;
+    return null;
   }
-  const project = userConfig.project || memoizedFindProject(folder);
+
+  const podfile = findPodfilePath(userConfig.sourceDir || folder);
 
   /**
-   * No iOS config found here
+   * In certain repos, the Xcode project can
+   * be generated by a tool. The only file that we can assume to exist on disk
+   * is `Podfile`.
    */
-  if (!project) {
+  if (!podfile) {
     return null;
   }
 
-  const projectPath = path.join(folder, project);
-  const podfile = findPodfilePath(folder, projectPath);
+  const sourceDir = path.dirname(podfile);
 
-  // This is a temporary fix for #1435. In certain repos, the Xcode project can
-  // be generated by a tool. The only file that we can assume to exist on disk
-  // is `Podfile`.
-  const sourceDir = podfile ? path.dirname(podfile) : path.dirname(projectPath);
+  /**
+   * @todo
+   * What if there are multiple `xcodeproj` found? Shall we read from a Podfile?
```

Comment:
Would it make sense to find the `.xcworkspace` instead? There should only be one of it (generated from running `pod install`).

Reply:
That function actually looks for `xcworkspace` in first place and returns `isWorkspace: true`. I guess in scenario when there's no `xcworkspace` but multiple projects, we can print a warning in verbose mode, similarly to `findPodfile`, asking users to look at custom config?

Reply:
Yeah, that sounds good to me.